DAGE 15: integration test

nicolo-rinaldi opened the pull request:
What i did in this ticket is to create a test file for all the search engines supported up to now (solr, elasticsearch and opensearch) → the only way to test in one go is to create one new docker-compose file (called docker-compose.full.ymlthat is used to extend all the other (in a future vespa integration, the vespa extension must be added). For each search engine there are 4 basic test function + one test for the full pipeline, that is contained in the file end_to_end_pipeline.py. It makes sense to move this test in a e2e/ folder inside tests/ but is going to be done (if needed) later on. 

For opensearch, I changed the port (9200 → 9300) since there were ports conflict due to the unification of the Docker container.

More than this, also a test with the interaction between the datastore and the writers has been added to the tests/integration/ folder. 

If the DataStore refactor on ticket [DAGE-37](https://sease.atlassian.net/browse/DAGE-37) is going to be merged before this, all the code that involves DataStore class must be changed (I think is only inside the latter tests I mentioned. On the opposite, there will be more code to solve conflict for Pablo in the Refactor.

[ticket](https://sease.atlassian.net/browse/DAGE-15)

nicolo-rinaldi committed:
[DAGE-15] update env and changed config file

nicolo-rinaldi committed:
[DAGE-15] created integration tests for docker init for each search engine (only solr working for now)

nicolo-rinaldi committed:
[DAGE-15] design of (not definitive) test for docker-compose

nicolo-rinaldi committed:
[DAGE-15] big bang integration test should be working now. More checks are needed

nicolo-rinaldi committed:
[DAGE-15] added last test e2e pipeline

nicolo-rinaldi committed:
[DAGE-15] modified last test e2e pipeline

nicolo-rinaldi committed:
[DAGE-15] bugfix on the docker container name that makes the tests crash

nicolo-rinaldi committed:
[DAGE-15] now solr alone works, but not with elastic

nicolo-rinaldi committed:
[DAGE-15] docker-related tests now works for openserach, elasticsearch and solr

nicolo-rinaldi committed:
[DAGE-15] added test file for the integration between the datastore and the writers, both rre and quepid (in file test_datastore_with_writer.py)

nicolo-rinaldi committed:
[DAGE-15] added check in docker until we reach the right number of documents (200) indexed inside the index

nicolo-rinaldi committed:
Update test.yml to handle separately integration and unit tests

nicolo-rinaldi committed:
Revert of the update of test.yml file

nicolo-rinaldi committed:
[DAGE-15] increased timeout for slow tests in github

nicolo-rinaldi committed:
[DAGE-15] Changed workflow file

Now the file is executing just unit tests

nicolo-rinaldi committed:
[DAGE-15] updated README.md with tests execution bash commands

Intrinsical-AI commented on rre-dataset-generator/tests/conftest.py:
```diff
@@ -0,0 +1,23 @@
+import subprocess
+import pytest
+
+@pytest.fixture(scope="session")
```
[Suggestion]

The conftest it's loaded for all the test files within that directoy. I think we could move this file to test/integration/ to avoid loading this when executing the unit (and the e2e in the future if we add it)

Intrinsical-AI commented on rre-dataset-generator/tests/integration/docker-compose.full.yml:
```diff
@@ -0,0 +1,36 @@
+  solr-init:
+    extends:
+      file: docker-compose.solr.yml
+      service: solr-init
```
Very clear and readable :+1:

Intrinsical-AI commented on rre-dataset-generator/tests/integration/end_to_end_pipeline.py:
```diff
@@ -0,0 +1,95 @@
+from src.writers.writer_factory import WriterFactory
+
+
+def end_to_end_pipeline_with_llm_mock(config, tmp_path):
```
[Notes]

1. If the file or the method does not start with "test_" pytest doesn't recognize it by default, so this has to be manually triggered (I think it's the intended behaviour)
2. I think is more like an e2e than integration test, but looks solid anw

nicolo-rinaldi committed:
[DAGE-15] first change to avoid repeating checks
Incoming needed change: trying to collapse all different search engines into one file (must change the search engines related classes)
